File: cms-user/src/main/java/cn/zmlio/microcms/entity/RoleEntity.java
package cn.zmlio.microcms.entity;

import lombok.Data;

import javax.persistence.*;
import java.util.Set;

@Entity
@Table(name = "t_roles")
public class RoleEntity extends BasicEntity {

    @Column(length = 64)
    private String roleCode;

    @Column(length = 20)
    private String roleName;

    @Column(length = 40)
    private String module;

    private boolean moduleDefault;

    @JoinTable(name = "t_role_access", joinColumns = {@JoinColumn(name = "role_id")},
            inverseJoinColumns = {@JoinColumn(name = "access_id")})
    @ManyToMany
    private Set<AccessEntity> accesses;

    @Override
    public String toString() {
        return "RoleEntity{" +
                "roleCode='" + roleCode + '\'' +
                ", roleName='" + roleName + '\'' +
                ", module='" + module + '\'' +
                ", moduleDefault=" + moduleDefault +
                "} " + super.toString();
    }

    public String getRoleCode() {
        return roleCode;
    }

    public void setRoleCode(String roleCode) {
        this.roleCode = roleCode;
    }

    public String getRoleName() {
        return roleName;
    }

    public void setRoleName(String roleName) {
        this.roleName = roleName;
    }

    public String getModule() {
        return module;
    }

    public void setModule(String module) {
        this.module = module;
    }

    public boolean isModuleDefault() {
        return moduleDefault;
    }

    public void setModuleDefault(boolean moduleDefault) {
        this.moduleDefault = moduleDefault;
    }

    public Set<AccessEntity> getAccesses() {
        return accesses;
    }

    public void setAccesses(Set<AccessEntity> accesses) {
        this.accesses = accesses;
    }
}

File: cms-user/src/main/java/cn/zmlio/microcms/entity/RoleEntity.kt
package cn.zmlio.microcms.entity

import javax.persistence.*

@Entity
@Table(name = "t_roles")
data class RoleEntity(@Column(length = 64)
                      var roleCode: String, @Column(length = 20)
                      var roleName: String,
                      @Column(length = 40)
                      var module: String,
                      var isModuleDefault: Boolean = false) : BasicEntity() {

    @JoinTable(name = "t_role_access", joinColumns = arrayOf(JoinColumn(name = "role_id")), inverseJoinColumns = arrayOf(JoinColumn(name = "access_id")))
    @ManyToMany
    var accesses: Set<AccessEntity>? = null

    override fun toString(): String {
        return "RoleEntity{" +
                "roleCode='" + roleCode + '\'' +
                ", roleName='" + roleName + '\'' +
                ", module='" + module + '\'' +
                ", moduleDefault=" + isModuleDefault +
                "} " + super.toString()
    }
}
